fastlane/fastlane
# Support custom s3 endpoint for Match storage

### Checklist

### Motivation and Context
Our company no longer exposes git repos (where we were storing mobile certs) to BitRise (where we are running `match` from). We can, however, store certs on our own instance of Minio, an open source S3, and expose it to BitRise.

### Description
This PR adds `fastlane match` options to support custom S3 endpoints.

### Testing Steps
Setup a local Minio server
```
brew install minio/stable/minio
minio server ~/minio_test_data
```
Create Minio bucket
```
brew install minio/stable/mc
mc alias set minio_test http://127.0.0.1:9000 minioadmin minioadmin
mc mb minio_test/mobile-certs
```
Configure a `Matchfile` like this: 
```ruby
storage_mode "s3"
s3_endpoint "http://127.0.0.1:9000" # new
s3_access_key "minioadmin"
s3_secret_access_key "minioadmin"
s3_bucket "mobile-certs"
s3_region "us-west-1"
s3_ssl_verify_peer false # new
s3_force_path_style true # new
```
Run match
```
bundle exec fastlane match
```
Check for files with `mc ls`:
```
mc ls minio_test/mobile-certs
```
